File: internal/content/testdata/bench/go/array-left-rotation.go
func left_rotate(arr []int, k int) []int {
	n := len(arr)
	if n == 0 {
		return arr
	}
	k = k % n
	result := make([]int, n)
	copy(result, arr[k:])
	copy(result[n-k:], arr[:k])
	return result
}

File: internal/content/testdata/bench/java/array-left-rotation.java
class Solution {
    public long[] left_rotate(long[] arr, long k) {
        int n = arr.length;
        if (n == 0) return arr;
        int shift = (int)(k % n);
        long[] result = new long[n];
        for (int i = 0; i < n; i++) {
            result[i] = arr[(i + shift) % n];
        }
        return result;
    }
}
